p2p/src/lib.rs

```diff
@@ -159,13 +162,15 @@ pub async fn make_p2p(
     let stream_adapter = NoiseEncryptionAdapter::gen_new();
     let base_transport = net::default_backend::transport::TcpTransportSocket::new();
     let transport = NoiseTcpTransport::new(stream_adapter, base_transport);
+    let peerdb_store = Box::new(PeerDbStoreImpl::new(storage::inmemory::InMemory::new())?);
```

Comment:
This should be initialized in a way similar to how chainstate storage is initialized, where it can depend on the data dir.

Reply:
Should be done. Default is the LMDB backend (until the Sqlite backend is ready) and unit tests must specify InMemory backend explicitly. Otherwise tests might interfere with each other.

Reply:
There's a misunderstanding here. There's no default other than what you provide. Obviously we shouldn't use lmdb for tests, but in runner.rs, this should use lmdb. In unit tests, we pass inmemory. In functional tests, lmdb will be used. Am I missing something?

Reply:
I updated that part. Should be OK now, thanks.